src/startApp.ts

```diff
@@ -88,13 +99,45 @@ async function loadEntryPointFile(functionAppDirectory: string): Promise<void> {
             }
         } catch (err) {
             const error = ensureErrorType(err);
-            worker.log({
-                message: `Worker was unable to load entry point "${currentFile ? currentFile : entryPointPattern}": ${
-                    error.message
-                }`,
-                level: LogLevel.Warning,
-                logCategory: LogCategory.System,
-            });
+            const newMessage = `Worker was unable to load entry point "${currentFile || entryPointPattern}": ${
+                error.message
+            }`;
+
+            if (shouldBlockOnEntryPointError()) {
+                error.message = newMessage;
+                error.isAzureFunctionsSystemError = true;
+                // We don't want to throw this error now (during workerInit or funcEnvReload) because technically the worker is fine
+                // Instead, it will be thrown during functionMetadata or functionLoad response which better indicates that the user's app is the problem
+                worker.app.blockingAppStartError = error;
+                // This will ensure the error makes it to the user's app insights
+                console.log(error.stack);
+            } else {
+                // In this case, the error will never block the app
+                // The most we can do without breaking backwards compatibility is log it as a system log
+                worker.log({
```

Comment:
Why don't we also use the `console.log` here?

Reply:
Because new errors showing up in a user's app insights can be considered a breaking change. Even if it doesn't change the behavior of the app, people still have alerts and stuff for errors being logged.

Reply:
Is it a breaking change if it's technically fixing a bug that those errors weren't being logged before? They aren't new errors, they were always there, they just weren't being logged to app insights. We had a similar precedent with #555 (though that did cause some problems for Durable users, so I don't have strong feelings about this)

Reply:
They are new errors (relatively) because we just added support for entry points last fall. At a higher level yeah unfortunately fixing a bug can often be a breaking change because people get used to the broken behavior

For better or worse, #555 never really affected users because of our various app insights problems with the worker. That PR allowed the logs to show up in our FunctionsLogs table for the first time, but they still don't show up in a user's app insights :(

Reply:
Makes sense. Like I said, I don't have strong feelings here, so happy to err on the side of caution